pkg/kubelet/cri/streaming/portforward/httpstream.go

```diff
@@ -63,7 +63,7 @@ func handleHTTPStreams(req *http.Request, w http.ResponseWriter, portForwarder P
 		uid:                   uid,
 		forwarder:             portForwarder,
 	}
-	h.run()
+	h.run(req.Context()) // TODO confirm this is correct
```

Comment:
This was added by me.

Reply:
this looks right... run blocks until we're done, at which point we close the connection in a defer, so request context lifetime seems appropriate

Reply:
I'm not sure req.Context is valid

>	conn := upgrader.UpgradeResponse(w, req, httpStreamReceived(streamChan)

this hijacks the connection

https://github.com/kubernetes/kubernetes/blob/00cd2ae3bc902f8c7a116e9602acf2b486a8f38f/staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/upgrade.go#L95-L106

so the request context should no longer be valid

Reply:
but, since run() is blocking, the request context should still be good for bailing out of the loop, right?

Reply:
good question, my point is that I don't know if the request context applies to the connection
There is some related issues in golang

https://github.com/golang/go/issues/22347
 https://go-review.googlesource.com/c/go/+/119235

Reply:
it seems that the comments agree with you Jordan, the context is valid

>  	// After a call to Hijack, the original Request.Body must not
> 	// be used. The original Request's Context remains valid and
> 	// is not canceled until the Request's ServeHTTP method
> 	// returns.